test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,311 @@
+				g.Expect(err).NotTo(gomega.HaveOccurred())
+				return ok
+			}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+			gomega.Expect(len(createdPod.Spec.NodeSelector)).Should(gomega.Equal(1))
+			gomega.Expect(createdPod.Spec.NodeSelector["kubernetes.io/arch"]).Should(gomega.Equal("arm64"))
+
+			gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(gomega.Succeed())
+			gomega.Expect(createdWorkload.Status.Conditions).Should(gomega.BeComparableTo(
+				[]metav1.Condition{
+					{Type: kueue.WorkloadQuotaReserved, Status: metav1.ConditionTrue},
+					{Type: kueue.WorkloadAdmitted, Status: metav1.ConditionTrue},
+				},
+				wlConditionCmpOpts...,
+			))
+
+			ginkgo.By("checking that pod is stopped when workload is evicted")
+
+			gomega.Expect(
+				workload.UpdateStatus(ctx, k8sClient, createdWorkload, kueue.WorkloadEvicted, metav1.ConditionTrue,
+					kueue.WorkloadEvictedByPreemption, "By test", "evict"),
+			).Should(gomega.Succeed())
+			util.FinishEvictionForWorkloads(ctx, k8sClient, createdWorkload)
+
+			gomega.Eventually(func(g gomega.Gomega) bool {
+				g.Expect(k8sClient.Get(ctx, lookupKey, createdPod)).To(gomega.Succeed())
+				return createdPod.DeletionTimestamp.IsZero()
+			}, util.Timeout, util.Interval).Should(gomega.BeFalse(), "Expected pod to be deleted")
+
+			gomega.Expect(createdPod.Status.Conditions).Should(gomega.ContainElement(corev1.PodCondition{
+				Type:    "TerminationTarget",
+				Status:  "True",
+				Reason:  "StoppedByKueue",
+				Message: "By test",
+			}))
+		})
+
+		ginkgo.When("Pod owner is managed by Kueue", func() {
+			var pod *corev1.Pod
+			ginkgo.BeforeEach(func() {
+				pod = testingpod.MakePod(podName, ns.Name).
+					Queue("test-queue").
+					OwnerReference("parent-job", batchv1.SchemeGroupVersion.WithKind("Job")).
+					Obj()
+			})
+
+			ginkgo.It("Should skip the pod", func() {
+				gomega.Expect(k8sClient.Create(ctx, pod)).Should(gomega.Succeed())
+
+				createdPod := &corev1.Pod{}
+				gomega.Eventually(func() error {
+					return k8sClient.Get(ctx, lookupKey, createdPod)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+				gomega.Expect(createdPod.Spec.SchedulingGates).NotTo(
+					gomega.ContainElement(corev1.PodSchedulingGate{Name: "kueue.x-k8s.io/admission"}),
+					"Pod shouldn't have scheduling gate",
+				)
+
+				gomega.Expect(createdPod.Labels).NotTo(
+					gomega.HaveKeyWithValue("kueue.x-k8s.io/managed", "true"),
+					"Pod shouldn't have the label",
+				)
+
+				gomega.Expect(createdPod.Finalizers).NotTo(gomega.ContainElement("kueue.x-k8s.io/managed"),
+					"Pod shouldn't have finalizer")
+
+				ginkgo.By(fmt.Sprintf("checking that workload '%s' is not created", wlLookupKey))
+				createdWorkload := &kueue.Workload{}
+
+				gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).To(testing.BeNotFoundError())
+			})
+		})
+	})
+})
```

Comment:
Add test cases when starting manager with the scheduler like other integrations:

https://github.com/kubernetes-sigs/kueue/blob/4e6592803902e2bc5db39a66ddf63e89ad129247/test/integration/controller/jobs/jobset/jobset_controller_test.go#L586